### PR TITLE
`Exam mode`: Hide quiz lifecycle buttons

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Exam exercises are handled by the exam itself and should not offer additional lifecycle actions like `start now`.

### Description

Removes the lifecycle buttons for quizzes in exams. Additionally verifies in the Rest endpoints that the exercises are not part of an exam if the lifecycle actions are invoked.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Create/Find an exam with a quiz exercise
3. Check that the detail page of that quiz does not show any buttons like "start now"

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots

Example with start button that should no longer be visible:
Bildschirmfoto 2024-07-30 um 11 32 36

## Summary by CodeRabbit

- **New Features**
	- Introduced restrictions on operations for quiz exercises designated as exam exercises, preventing users from joining, creating, or starting quiz batches.
	- Conditional rendering of lifecycle buttons based on quiz exercise mode in the quiz exercise detail view.

- **Enhancements**
	- Added new error messages for restricted operations in both German and English localizations, improving user feedback during exam scenarios.

- **Tests**
	- Expanded testing coverage for quiz exercises, validating expected error responses for invalid operations related to exam exercises.

### CODE REVIEW

**Review by pzdr7 — DISMISSED** [review]
Code

File: src/main/webapp/i18n/de/error.json [hunk 1]
```diff
@@ -94,6 +94,7 @@
         "unableToImportProgrammingExercise": "Fehler beim Importieren der Programmieraufgabe.",
         "titleAlreadyExists": "Es existiert bereits eine Programmieraufgabe mit demselben Titel, bitte nutze einen anderen Titel.",
         "shortnameAlreadyExists": "Es existiert bereits eine Programmieraufgabe mit demselben Kurznamen, bitte nutze einen anderen Kurznamen.",
-        "courseTitleTooLong": "Der Kurstitel ist zu lang (max. 255 Zeichen), bitte nutze einen kürzeren Kurstitel."
+        "courseTitleTooLong": "Der Kurstitel ist zu lang (max. 255 Zeichen), bitte nutze einen kürzeren Kurstitel.",
+        "notAllowedInExam": "Diese Operation ist in einer Klausur nicht erlaubt."
```
Comment (anian03) [inline src/main/webapp/i18n/de/error.json:98]: This sounds a bit strange to me, I don't think "operation" translates like that. Maybe use "Aktion" instead
```suggestion
        "notAllowedInExam": "Diese Aktion ist in einer Klausur nicht erlaubt."
```

**Review by rabeatwork — DISMISSED** [review]

**Review by BBesrour — DISMISSED** [review]
code

**Review by pzdr7 — APPROVED** [review]
Reapprove

**Review by anian03 — APPROVED** [review]

**Review by BBesrour — APPROVED** [review]
tested locally, lgtm

**Review by EneaGore — APPROVED** [review]
Tested on TS1. Works as expected

**Review by rabeatwork — APPROVED** [review]

**Review by krusche — APPROVED** [review]
Code looks good 👍

**Pull request merged**